### PR TITLE
helm: Add priorityClass & nodeSelector to certgen jobs

### PR DESCRIPTION
**Pull request by adberger**
Please ensure your pull request adheres to the following guidelines:

- [x] For first time contributors, read [Submitting a pull request](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#submitting-a-pull-request)
- [ ] All code is covered by unit and/or runtime tests where feasible.
- [x] All commits contain a well written commit description including a title,
      description and a `Fixes: #XXX` line if the commit addresses a particular
      GitHub issue.
- [ ] If your commit description contains a `Fixes: <commit-id>` tag, then
      please add the commit author[s] as reviewer[s] to this issue.
- [x] All commits are signed off. See the section [Developer’s Certificate of Origin](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#dev-coo)
- [ ] Provide a title or release-note blurb suitable for the release notes.
- [ ] Are you a user of Cilium? Please add yourself to the [Users doc](https://github.com/cilium/cilium/blob/main/USERS.md)
- [x] Thanks for contributing!

Currently the Helmchart to install Cilium does not allow to set a `priorityClassName` & `nodeSelector` for the certgen jobs.
This PR intends to fix this.

```release-note
helm: Add priorityClass & nodeSelector to certgen jobs
```

### CODE REVIEW

**Review by gandro — APPROVED** [review]
Thank you!

File: install/kubernetes/cilium/values.yaml.tmpl [hunk 1]
```diff
@@ -1098,6 +1098,10 @@ certgen:
   annotations:
     job: {}
     cronJob: {}
+  # ref: ref: https://kubernetes.io/docs/concepts/scheduling-eviction/assign-pod-node/#nodeselector
```
Comment (giorio94) [inline install/kubernetes/cilium/values.yaml.tmpl]:
```suggestion
  # Node selector for certgen pods
  # ref: https://kubernetes.io/docs/concepts/scheduling-eviction/assign-pod-node/#nodeselector
```

**Review by giorio94 — APPROVED** [review]
Thanks! Just one small nit inline.

File: install/kubernetes/cilium/values.yaml.tmpl (hunk 1 above)
Comment (adberger) [reply install/kubernetes/cilium/values.yaml.tmpl]: Thanks for the catch!
I've made some more changes, to align with the other values.

**Comment by giorio94** [comment]
Some automatically generated files [seem to be out of sync](https://github.com/cilium/cilium/actions/runs/11435620407/job/31811375315?pr=35429). You can fix that with:

```
make -C install/kubernetes
make -C Documentation update-helm-values 
```

**Comment by adberger** [comment]
> Some automatically generated files [seem to be out of sync](https://github.com/cilium/cilium/actions/runs/11435620407/job/31811375315?pr=35429). You can fix that with:
> 
> ```
> make -C install/kubernetes
> make -C Documentation update-helm-values 
> ```

Hmm, unfortunately I get:
```
make -C install/kubernetes
: cilium/Chart.yaml
grep -lR -e 'version:' -e 'appVersion:' /Users/adberger/cilium/install/kubernetes/cilium/ \
		| xargs -L 1 sed -i -e 's/''\([vV]ersion:\) ''[0-9]\+\.[0-9]\+\.[0-9]\+.*''/\1 '1.17.0-dev'/g'
sed -i 's;icon:.*;icon: https://cdn.jsdelivr.net/gh/cilium/cilium@main/Documentation/images/logo-solo.svg;' "/Users/adberger/cilium/install/kubernetes/cilium/Chart.yaml"
sed: 1: "/Users/adberger/cilium ...": command a expects \ followed by text
make: *** [update-chart] Error 1
```

**Pull request merged**